### ISSUE TITLE
Show branch in search results

### ISSUE THREAD

**Issue by HubMiner** [osmandapp/Osmand#6947]
Enhancement request:

When searching for a common name (ex: Starbucks) and search results contain "branch" tag, please show that in the results, possible format: Name - Branch - Distance

https://wiki.openstreetmap.org/wiki/Key%3Abranch (wiki exists since 2016).
https://taginfo.openstreetmap.org/keys/branch#overview

Thank you!

**Comment by HubMiner** [comment]
Additional idea: if branch tag is not available, but addr:city is, show that in search results:

Name - City - Distance

**Comment by aceman444** [comment]
Yes, notice OsmAnd is specifically mentioned on the wiki page as a possible consumer of this information.

**Comment by dpriskorn** [comment]
See https://github.com/openstreetmap/iD/issues/8856 which is related to this UI deficiency

**Comment by dpriskorn** [comment]
@vshcherb I would like to try working on this. Where do I start? Which IDE do you recommend?

**Comment by vshcherb** [comment]
Interesting... Yes, you could start to add as UI, I think you're correct (cause I don't know for sure ). Android Studio - https://docs.osmand.net/en/main@latest/development/build-osmand/how-to-compile-the-android-version

**Comment by dpriskorn** [comment]
> Interesting... Yes, you could start to add as UI, I think you're correct (cause I don't know for sure ). Android Studio - https://docs.osmand.net/en/main@latest/development/build-osmand/how-to-compile-the-android-version

Thanks 😃 I tried the repo tool but running "repo sync" on Win10 in GitBash did not work 🤷‍♂️
I cloned manually and tried building the master branch, but hit a bug it seems. I wrote asking for help in the english Telegram channel.

**Comment by jfburke3** [comment]
I want to plus5 this.  Just to cite some examples from personal experience, when searching for some examples, knowing the branch location makes it much easier to find the best location to go to.  I honestly think this would make OSMAND a (small) step ahead of other nav apps.

**Comment by EvanCarroll** [comment]
This would greatly improve usability. When you're in the middle of a city -- downtown, and someone says book a stay at the Holiday Inn Westchase, which is in Houston. It's impossible to find. There are like 5 Holidays Inns equal distance from the center center spread in all directions.

**Comment by EvanCarroll** [comment]
@vshcherb I think you should put this one ahead of #2993 on 4.4-backend. It's far more valuable to the user. Though I think falling back to city is a good idea.

**Comment by vshcherb** [comment]
I think it's important issue to implement but we will be very lucky if it happens this year, likely next year

**Comment by gena01** [comment]
I am running into this one more and more as we have ton of "brand" stores a few miles apart. This is actually a bigger challenge when driving and using navigation. One scenario i ran into recently was a dilemma picking a branch I was driving  to vs the one behind me (both relatively close from where I was and both located on the same road).

P.S. As an osm contributor/mapper it would also greatly help me visualize and see that "Branch"es for big brand(s) are tagged properly.

**Comment by pomeloRHO** [comment]
This is important when looking for places when you need to find the specific branch is. For example: when looking for a clinic in your health society, all results appear with the same name but you need to go to the one where your schedule is.
